src/cozmo/conn.py

```diff
@@ -299,6 +299,24 @@ def _handle_ui_device_connected(self, msg):
         build_versions_match = (cozmoclad.__build_version__ == '00000.00000.00000'
             or cozmoclad.__build_version__ == msg.buildVersion)
 
+        if clad_hashes_match and not build_versions_match:
+            # If CLAD hashes match, and this is only a minor version change, allow connection but recommend user upgrades
+            sdk_major_version = cozmoclad.__build_version__.split(".")[0:2]
+            build_major_version = msg.buildVersion.split(".")[0:2]
+            build_versions_match = (sdk_major_version == build_major_version)
+            if build_versions_match:
+                if cozmoclad.__build_version__ < msg.buildVersion:
+                    # App is newer
+                    logger.warning(
+                        'Your App is newer than your SDK: '
+                        'We recommend that you upgrade by calling command: '
+                        '"pip3 install --user --ignore-installed cozmo" '
+                        'and downloading the latest examples from http://cozmosdk.anki.com/docs/downloads.html')
+                else:
+                    logger.warning(
+                        'Your App is older than your SDK: '
+                        'We recommend that you download the latest app from the app store.')
```

Comment:
In "the latest app", app should be App to match the other references to App in this new code.